plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/flexible/FlexDeployPreferencesPanel.java

```diff
@@ -38,20 +39,25 @@
   public FlexDeployPreferencesPanel(Composite parent, IProject project,
       IGoogleLoginService loginService, Runnable layoutChangedHandler, boolean requireValues,
       ProjectRepository projectRepository) {
-    super(parent, project, loginService, layoutChangedHandler, requireValues, projectRepository,
+    this(parent, project, loginService, layoutChangedHandler, requireValues, projectRepository,
         new FlexDeployPreferences(project));
   }
 
+  protected FlexDeployPreferencesPanel(Composite parent, IProject project,
+      IGoogleLoginService loginService, Runnable layoutChangedHandler, boolean requireValues,
+      ProjectRepository projectRepository, DeployPreferences model) {
+    super(parent, project, loginService, layoutChangedHandler, requireValues, projectRepository,
+        model);
+  }
+
   @Override
   protected void createCenterArea() {
-    IPath fileFieldBasePath = project.getLocation();
-
     Text appYamlField = createBrowseFileRow(
         Messages.getString("deploy.preferences.dialog.label.app.yaml"),
         Messages.getString("tooltip.app.yaml"),
-        fileFieldBasePath, new String[] {"*.yaml"});
+        getFileFieldBasePath(), new String[] {"*.yaml"});
```

Comment:
This method name ` getFileFieldBasePath` isn't really obvious to me. Any other ideas?

Reply:
I can add a Javadoc.

Reply:
I'd rather rename so it's more obvious. I completely misunderstood what this did on first read.

Reply:
`getDialogBasePathForFileFields`?

Reply:
That's going in the wrong direction. :-)  Try not to reference the UI here. E.g. `getWorkingDirectory()`

Reply:
It's not a working directory for something, but a path that will be automatically added as a prefix if users enter a non-absolute path in file input fields in the dialog. For example, if a user enters `src/main/appengine/app.yaml` and this base path is an absolute path of a project `/home/user/workspace/my-project`, we can construct `/home/user/workspace/my-project/src/main/appengine/app.yaml`.

Reply:
On second thought, `getWorkingDirectory()` is good enough.